Error: Plugin did not respond

From 08/06/2021 running Terraform on Azure DevOps agents
Tried with Terraform version 15.5 and 15.1
On multiple pipelines getting same errors

unable to complete builds

tested and Terraform is working from local machine so this is just affecting Azure DevOps agents

How can I get the plugin logs exported from the DevOps Agent??
Raw Error Log:

2021-06-09T09:17:56.4744163Z ##[section]Starting: Terraform
2021-06-09T09:17:56.4750811Z ==============================================================================
2021-06-09T09:17:56.4751166Z Task : Command line
2021-06-09T09:17:56.4751525Z Description : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
2021-06-09T09:17:56.4751874Z Version : 2.182.0
2021-06-09T09:17:56.4752155Z Author : Microsoft Corporation
2021-06-09T09:17:56.4752541Z Help : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
2021-06-09T09:17:56.4752963Z ==============================================================================
2021-06-09T09:17:56.5923484Z Generating script.
2021-06-09T09:17:56.5932081Z Script contents:
2021-06-09T09:17:56.5933584Z terraform apply -auto-approve -input=false tfplan
2021-06-09T09:17:56.5934098Z ========================== Starting Command Output ===========================
2021-06-09T09:17:56.5952091Z [command]/usr/bin/bash --noprofile --norc /home/vsts/work/_temp/d47a6dbe-1bba-4523-bc12-41efffe87256.sh
2021-06-09T09:18:02.4256807Z �[0m�[1mazurerm_resource_group.AzureResourceGroup: Creating...�[0m�[0m
2021-06-09T09:18:02.7613304Z �[0m�[1mazurerm_resource_group.AzureResourceGroup: Creation complete after 1s [id=/subscriptions/***/resourceGroups/ASM0062-PLAN-UKS-NEW2-RGNew]�[0m
2021-06-09T09:18:02.7681756Z �[0m�[1mazurerm_public_ip.AzurePublicIP[1]: Creating...�[0m�[0m
2021-06-09T09:18:02.7682669Z �[0m�[1mazurerm_local_network_gateway.LNG: Creating...�[0m�[0m
2021-06-09T09:18:02.7683436Z �[0m�[1mazurerm_key_vault.TerraformAzureKeyVault: Creating...�[0m�[0m
2021-06-09T09:18:02.7684197Z �[0m�[1mazurerm_recovery_services_vault.RSV: Creating...�[0m�[0m
2021-06-09T09:18:02.7684940Z �[0m�[1mazurerm_virtual_network.AzureNetwork: Creating...�[0m�[0m
2021-06-09T09:18:02.7736209Z �[0m�[1mazurerm_public_ip.AzurePublicIP[0]: Creating...�[0m�[0m
2021-06-09T09:18:02.7765153Z �[0m�[1mazurerm_storage_account.AzureStorageAccount[0]: Creating...�[0m�[0m
2021-06-09T09:18:02.7774386Z �[0m�[1mazurerm_public_ip.VNGWPublicIP: Creating...�[0m�[0m
2021-06-09T09:18:02.8890279Z �[31m╷�[0m�[0m
2021-06-09T09:18:02.8891610Z �[31m│�[0m �[0m�[1m�[31mError: �[0m�[0m�[1mPlugin did not respond�[0m
2021-06-09T09:18:02.8892460Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8893359Z �[31m│�[0m �[0m�[0m with azurerm_key_vault.TerraformAzureKeyVault,
2021-06-09T09:18:02.8894267Z �[31m│�[0m �[0m on AzureKeyVault.tf line 56, in resource "azurerm_key_vault" "TerraformAzureKeyVault":
2021-06-09T09:18:02.8895293Z �[31m│�[0m �[0m 56: resource "azurerm_key_vault" "TerraformAzureKeyVault" �[4m{�[0m�[0m
2021-06-09T09:18:02.8896160Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8896947Z �[31m│�[0m �[0mThe plugin encountered an error, and failed to respond to the
2021-06-09T09:18:02.8897851Z �[31m│�[0m �[0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2021-06-09T09:18:02.8898836Z �[31m│�[0m �[0mcontain more details.
2021-06-09T09:18:02.8902006Z �[31m╵�[0m�[0m
2021-06-09T09:18:02.8902873Z �[31m╷�[0m�[0m
2021-06-09T09:18:02.8903928Z �[31m│�[0m �[0m�[1m�[31mError: �[0m�[0m�[1mPlugin did not respond�[0m
2021-06-09T09:18:02.8904697Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8905448Z �[31m│�[0m �[0m�[0m with azurerm_virtual_network.AzureNetwork,
2021-06-09T09:18:02.8906419Z �[31m│�[0m �[0m on AzureNetwork.tf line 2, in resource "azurerm_virtual_network" "AzureNetwork":
2021-06-09T09:18:02.8909109Z �[31m│�[0m �[0m 2: resource "azurerm_virtual_network" "AzureNetwork" �[4m{�[0m�[0m
2021-06-09T09:18:02.8909959Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8910747Z �[31m│�[0m �[0mThe plugin encountered an error, and failed to respond to the
2021-06-09T09:18:02.8911683Z �[31m│�[0m �[0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2021-06-09T09:18:02.8912564Z �[31m│�[0m �[0mcontain more details.
2021-06-09T09:18:02.8913322Z �[31m╵�[0m�[0m
2021-06-09T09:18:02.8915598Z �[31m╷�[0m�[0m
2021-06-09T09:18:02.8916555Z �[31m│�[0m �[0m�[1m�[31mError: �[0m�[0m�[1mPlugin did not respond�[0m
2021-06-09T09:18:02.8917380Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8918495Z �[31m│�[0m �[0m�[0m with azurerm_public_ip.AzurePublicIP[1],
2021-06-09T09:18:02.8919545Z �[31m│�[0m �[0m on AzureNetwork.tf line 26, in resource "azurerm_public_ip" "AzurePublicIP":
2021-06-09T09:18:02.8920674Z �[31m│�[0m �[0m 26: resource "azurerm_public_ip" "AzurePublicIP" �[4m{�[0m�[0m
2021-06-09T09:18:02.8921513Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8922301Z �[31m│�[0m �[0mThe plugin encountered an error, and failed to respond to the
2021-06-09T09:18:02.8923217Z �[31m│�[0m �[0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2021-06-09T09:18:02.8924085Z �[31m│�[0m �[0mcontain more details.
2021-06-09T09:18:02.8924774Z �[31m╵�[0m�[0m
2021-06-09T09:18:02.8925481Z �[31m╷�[0m�[0m
2021-06-09T09:18:02.8926193Z �[31m│�[0m �[0m�[1m�[31mError: �[0m�[0m�[1mPlugin did not respond�[0m
2021-06-09T09:18:02.8927044Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8927725Z �[31m│�[0m �[0m�[0m with azurerm_public_ip.AzurePublicIP[0],
2021-06-09T09:18:02.8928588Z �[31m│�[0m �[0m on AzureNetwork.tf line 26, in resource "azurerm_public_ip" "AzurePublicIP":
2021-06-09T09:18:02.8929672Z �[31m│�[0m �[0m 26: resource "azurerm_public_ip" "AzurePublicIP" �[4m{�[0m�[0m
2021-06-09T09:18:02.8930339Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8931041Z �[31m│�[0m �[0mThe plugin encountered an error, and failed to respond to the
2021-06-09T09:18:02.8931849Z �[31m│�[0m �[0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2021-06-09T09:18:02.8932548Z �[31m│�[0m �[0mcontain more details.
2021-06-09T09:18:02.8933306Z �[31m╵�[0m�[0m
2021-06-09T09:18:02.8959818Z �[31m╷�[0m�[0m
2021-06-09T09:18:02.8960695Z �[31m│�[0m �[0m�[1m�[31mError: �[0m�[0m�[1mPlugin did not respond�[0m
2021-06-09T09:18:02.8961363Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8962026Z �[31m│�[0m �[0m�[0m with azurerm_public_ip.VNGWPublicIP,
2021-06-09T09:18:02.8963095Z �[31m│�[0m �[0m on AzureNetworkVPN.tf line 4, in resource "azurerm_public_ip" "VNGWPublicIP":
2021-06-09T09:18:02.8963990Z �[31m│�[0m �[0m 4: resource "azurerm_public_ip" "VNGWPublicIP" �[4m{�[0m�[0m
2021-06-09T09:18:02.8964690Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8965387Z �[31m│�[0m �[0mThe plugin encountered an error, and failed to respond to the
2021-06-09T09:18:02.8966176Z �[31m│�[0m �[0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2021-06-09T09:18:02.8966898Z �[31m│�[0m �[0mcontain more details.
2021-06-09T09:18:02.8967494Z �[31m╵�[0m�[0m
2021-06-09T09:18:02.8970388Z �[31m╷�[0m�[0m
2021-06-09T09:18:02.8971172Z �[31m│�[0m �[0m�[1m�[31mError: �[0m�[0m�[1mPlugin did not respond�[0m
2021-06-09T09:18:02.8971833Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8972485Z �[31m│�[0m �[0m�[0m with azurerm_local_network_gateway.LNG,
2021-06-09T09:18:02.8973362Z �[31m│�[0m �[0m on AzureNetworkVPN.tf line 52, in resource "azurerm_local_network_gateway" "LNG":
2021-06-09T09:18:02.8975692Z �[31m│�[0m �[0m 52: resource "azurerm_local_network_gateway" "LNG" �[4m{�[0m�[0m
2021-06-09T09:18:02.8976402Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8977124Z �[31m│�[0m �[0mThe plugin encountered an error, and failed to respond to the
2021-06-09T09:18:02.8977969Z �[31m│�[0m �[0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2021-06-09T09:18:02.8980028Z �[31m│�[0m �[0mcontain more details.
2021-06-09T09:18:02.8980690Z �[31m╵�[0m�[0m
2021-06-09T09:18:02.8981293Z �[31m╷�[0m�[0m
2021-06-09T09:18:02.8982010Z �[31m│�[0m �[0m�[1m�[31mError: �[0m�[0m�[1mPlugin did not respond�[0m
2021-06-09T09:18:02.8982651Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8983318Z �[31m│�[0m �[0m�[0m with azurerm_recovery_services_vault.RSV,
2021-06-09T09:18:02.8984401Z �[31m│�[0m �[0m on AzureRSV.tf line 2, in resource "azurerm_recovery_services_vault" "RSV":
2021-06-09T09:18:02.8985331Z �[31m│�[0m �[0m 2: resource "azurerm_recovery_services_vault" "RSV" �[4m{�[0m�[0m
2021-06-09T09:18:02.8986002Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8986766Z �[31m│�[0m �[0mThe plugin encountered an error, and failed to respond to the
2021-06-09T09:18:02.8987551Z �[31m│�[0m �[0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2021-06-09T09:18:02.8988449Z �[31m│�[0m �[0mcontain more details.
2021-06-09T09:18:02.8989038Z �[31m╵�[0m�[0m
2021-06-09T09:18:02.8989589Z �[31m╷�[0m�[0m
2021-06-09T09:18:02.8990296Z �[31m│�[0m �[0m�[1m�[31mError: �[0m�[0m�[1mPlugin did not respond�[0m
2021-06-09T09:18:02.8990938Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8991627Z �[31m│�[0m �[0m�[0m with azurerm_storage_account.AzureStorageAccount[0],
2021-06-09T09:18:02.8992551Z �[31m│�[0m �[0m on AzureStorageAccount.tf line 2, in resource "azurerm_storage_account" "AzureStorageAccount":
2021-06-09T09:18:02.8993503Z �[31m│�[0m �[0m 2: resource "azurerm_storage_account" "AzureStorageAccount" �[4m{�[0m�[0m
2021-06-09T09:18:02.8994174Z �[31m│�[0m �[0m
2021-06-09T09:18:02.8994865Z �[31m│�[0m �[0mThe plugin encountered an error, and failed to respond to the
2021-06-09T09:18:02.8995716Z �[31m│�[0m �[0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2021-06-09T09:18:02.8996427Z �[31m│�[0m �[0mcontain more details.
2021-06-09T09:18:02.8997018Z �[31m╵�[0m�[0m
2021-06-09T09:18:02.9333530Z
2021-06-09T09:18:02.9334946Z Stack trace from the terraform-provider-azurerm_v2.62.1_x5 plugin:
2021-06-09T09:18:02.9335564Z
2021-06-09T09:18:02.9336472Z panic: interface conversion: interface {} is nil, not string
2021-06-09T09:18:02.9336779Z
2021-06-09T09:18:02.9337158Z goroutine 155 [running]:
2021-06-09T09:18:02.9338400Z github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network.expandLocalNetworkGatewayAddressSpaces(0xc001117b20, 0x5c4107d)
2021-06-09T09:18:02.9339641Z /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/local_network_gateway_resource.go:278 +0x20c
2021-06-09T09:18:02.9340934Z github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network.resourceLocalNetworkGatewayCreateUpdate(0xc001117b20, 0x545cf20, 0xc001443180, 0x0, 0x0)
2021-06-09T09:18:02.9342463Z /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/local_network_gateway_resource.go:149 +0x4e6
2021-06-09T09:18:02.9343971Z github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000b3df40, 0xc001a84140, 0xc00194b800, 0x545cf20, 0xc001443180, 0x548a601, 0xc0019ece38, 0xc001a873b0)
2021-06-09T09:18:02.9345289Z /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:320 +0x375
2021-06-09T09:18:02.9346550Z github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000182a00, 0xc001aa3a38, 0xc001a84140, 0xc00194b800, 0xc0019ece80, 0xc0019f4930, 0x548cbe0)
2021-06-09T09:18:02.9347829Z /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:294 +0x99
2021-06-09T09:18:02.9349164Z github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc000136378, 0x63544b0, 0xc0019cbc80, 0xc001117650, 0xc000136378, 0xc0019cbc80, 0xc00125dba0)
2021-06-09T09:18:02.9350526Z /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:895 +0x8a5
2021-06-09T09:18:02.9351930Z github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x5945a80, 0xc000136378, 0x63544b0, 0xc0019cbc80, 0xc0019e8840, 0x0, 0x63544b0, 0xc0019cbc80, 0xc001a80000, 0x29d)
2021-06-09T09:18:02.9353302Z /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
2021-06-09T09:18:02.9354207Z google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002116c0, 0x639a398, 0xc0003ad380, 0xc001c7e200, 0xc000be9680, 0x9785a80, 0x0, 0x0, 0x0)
2021-06-09T09:18:02.9355435Z /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1194 +0x52b
2021-06-09T09:18:02.9356206Z google.golang.org/grpc.(*Server).handleStream(0xc0002116c0, 0x639a398, 0xc0003ad380, 0xc001c7e200, 0x0)
2021-06-09T09:18:02.9373445Z /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1517 +0xd0c
2021-06-09T09:18:02.9374282Z google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00110dc20, 0xc0002116c0, 0x639a398, 0xc0003ad380, 0xc001c7e200)
2021-06-09T09:18:02.9375365Z /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:859 +0xab
2021-06-09T09:18:02.9376062Z created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-06-09T09:18:02.9377031Z /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:857 +0x1fd
2021-06-09T09:18:02.9377446Z
2021-06-09T09:18:02.9378045Z Error: The terraform-provider-azurerm_v2.62.1_x5 plugin crashed!
2021-06-09T09:18:02.9378293Z
2021-06-09T09:18:02.9378736Z This is always indicative of a bug within the plugin. It would be immensely
2021-06-09T09:18:02.9379484Z helpful if you could report the crash with the plugin's maintainers so that it
2021-06-09T09:18:02.9380028Z can be fixed. The output above should help diagnose the issue.
2021-06-09T09:18:02.9380389Z
2021-06-09T09:18:02.9438905Z ##[error]Bash exited with code '1'.
2021-06-09T09:18:02.9452491Z ##[section]Finishing: Terraform

We are also running into this same issue.

```

Stack trace from the terraform-provider-azurerm_v2.57.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x4c06da7]

goroutine 90 [running]:
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/postgres.resourcePostgreSQLDatabaseDelete(0xc0007fe8c0, 0x52a93e0, 0xc000352900, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/postgres/postgresql_database_resource.go:171 +0x247
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc00083ff40, 0xc0017645f0, 0xc001762440, 0x52a93e0, 0xc000352900, 0x40e101, 0xc0012ed560, 0x561e7e0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:293 +0x4a6
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc0000af500, 0xc001269a38, 0xc0017645f0, 0xc001762440, 0x94a1640, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00000fc68, 0x613e7b0, 0xc0012d22d0, 0xc0007fe150, 0xc00000fc68, 0xc0012d22d0, 0xc001297ba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:895 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x5770cc0, 0xc00000fc68, 0x613e7b0, 0xc0012d22d0, 0xc0012dc0c0, 0x0, 0x613e7b0, 0xc0012d22d0, 0xc000438000, 0x217)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000238c40, 0x61822d8, 0xc000533080, 0xc001b0a100, 0xc000c1e4e0, 0x945f1e0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000238c40, 0x61822d8, 0xc000533080, 0xc001b0a100, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000b56b80, 0xc000238c40, 0x61822d8, 0xc000533080, 0xc001b0a100)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:857 +0x1fd

Error: The terraform-provider-azurerm_v2.57.0_x5 plugin crashed!

```

--- a/azurerm/internal/services/network/local_network_gateway_resource.go
+++ b/azurerm/internal/services/network/local_network_gateway_resource.go
@@ -146,7 +146,7 @@ func resourceLocalNetworkGatewayCreateUpdate(d *pluginsdk.ResourceData, meta int
 			return fmt.Errorf("error waiting for completion of Local Network Gateway %q (Resource Group %q): %+v", name, resGroup, err)
 		}
 	}
-	gateway.LocalNetworkGatewayPropertiesFormat.LocalNetworkAddressSpace.AddressPrefixes = expandLocalNetworkGatewayAddressSpaces(d)
+	gateway.LocalNetworkGatewayPropertiesFormat.LocalNetworkAddressSpace = expandLocalNetworkGatewayAddressSpaces(d)
 
 	future, err := client.CreateOrUpdate(ctx, resGroup, name, gateway)
 	if err != nil {
@@ -271,14 +271,16 @@ func expandLocalNetworkGatewayBGPSettings(d *pluginsdk.ResourceData) *network.Bg
 	return &bgpSettings
 }
 
-func expandLocalNetworkGatewayAddressSpaces(d *pluginsdk.ResourceData) *[]string {
+func expandLocalNetworkGatewayAddressSpaces(d *pluginsdk.ResourceData) *network.AddressSpace {
 	prefixes := make([]string, 0)
 
 	for _, pref := range d.Get("address_space").([]interface{}) {
 		prefixes = append(prefixes, pref.(string))
 	}
 
-	return &prefixes
+	return &network.AddressSpace{
+		AddressPrefixes: &prefixes,
+	}
 }
 
 func flattenLocalNetworkGatewayBGPSettings(input *network.BgpSettings) []interface{} {

